### PR TITLE
Add flags to build multiarch Go SDK containers and fix Go Dataflow ARM github actions test

### PR DESCRIPTION
**Pull request by celeste-zeng**
- Introduce new flags to build and push multi-arch Go containers(will not affect the current building of x86 containers if flags are not specified during build);
- Change the validatesRunner script so it builds and pushes multi-arch Go container;
- Fix Go Dataflow ARM github actions test by adding the missing $USER variable.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/go/test/run_validatesrunner_tests.sh [hunk 1]
```diff
@@ -430,8 +424,7 @@ echo ">>> RUNNING $RUNNER integration tests with pipeline options: $ARGS"
 cd ../..
 
 if [[ "$RUNNER" == "dataflow" ]]; then
-  # Delete the container locally and remotely
-  docker rmi $CONTAINER:$TAG || echo "Failed to remove container"
+  # Delete the container remotely
```
Comment (tvalentyn) [inline sdks/go/test/run_validatesrunner_tests.sh:427]: we could still try delete locally as best effort, if that would help non-arm path.

File: build.gradle.kts [hunk 2]
```diff
@@ -440,10 +440,6 @@ tasks.register("goPortablePreCommit") {
   dependsOn(":sdks:go:test:ulrValidatesRunner")
 }
 
-tasks.register("goPostCommitDataflowARM") {
```
Comment (tvalentyn) [inline build.gradle.kts:443]: I think i suggested to remove it before offline, but feel free to keep this, looks like it's a convention that other `go` tests follow even if there is dependency only on one target.

File: sdks/go/container/build.gradle [hunk 3]
```diff
@@ -38,8 +41,10 @@ docker {
   files "./build/"
   buildArgs(['pull_licenses': project.rootProject.hasProperty(["docker-pull-licenses"]) ||
                      project.rootProject.hasProperty(["isRelease"])])
-  buildx project.containerPlatforms() != [project.nativeArchitecture()]
+  buildx useBuildx
   platform(*project.containerPlatforms())
+  load useBuildx && !pushContainers
```
Comment (tvalentyn) [inline sdks/go/container/build.gradle:46]: I am not sure how useful it is to keep the image designed for a different platform. we should test and see that all layers (incl base) would be selected for the target platform after such loading + manual push to a different repo.

**Review by tvalentyn — APPROVED** [review]

**Review by tvalentyn — APPROVED** [review]

**Review by tvalentyn — APPROVED** [review]

File: sdks/go/test/run_validatesrunner_tests.sh (hunk 1 above)
Comment (celeste-zeng) [reply sdks/go/test/run_validatesrunner_tests.sh:427]: If there is no local container found, keep this line will cause the task to fail with the message saying "Failed to remove container". Is there a way to try to clean the container locally, but if not found, we only echo the message but no error thrown?

File: build.gradle.kts (hunk 2 above)
Comment (celeste-zeng) [reply build.gradle.kts:443]: @lostluck thoughts?

File: sdks/go/test/run_validatesrunner_tests.sh (hunk 1 above)
Comment (tvalentyn) [reply sdks/go/test/run_validatesrunner_tests.sh:427]: that is what should be happening there is an OR (||), so the command shouldn't fail, only error message is shown. We could adjust the error message text though.

Comment (celeste-zeng) [reply sdks/go/test/run_validatesrunner_tests.sh]: I see! Just changed the error message.

File: build.gradle.kts (hunk 2 above)
Comment (lostluck) [reply build.gradle.kts:443]: I have no opinion on gradle. Follow the conventions that exist.

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
I can merge when tests pass.

**Comment by lostluck** [comment]
This PR broke the go postcommits, which were not run to validate the change. Triggering a revert.

**Comment by lostluck** [comment]
Which it can't do cleanly. Great.

**Comment by lostluck** [comment]
#27747 should be fixing this. Filed #27748 as a way to possibly prevent this from happening (generally with arbitrary post commits issues, perhaps not with a jenkins/GH mismatch)

**Pull request merged**